src/interpreter/example-validator.ts

```diff
@@ -0,0 +1,482 @@
+      case 'ListDefinition':
+        return this.visitListDefinitionNode(node);
+      case 'EnumDefinition':
+        return this.visitEnumDefinitionNode(node);
+      case 'EnumValue':
+        return this.visitEnumValueNode(node);
+      case 'PrimitiveTypeName':
+        return this.visitPrimitiveTypeNameNode(node);
+      default:
+        assertUnreachable(node);
+    }
+  }
+
+  visitProfileDocumentNode(node: ProfileDocumentNode): void {
+    const useCases = node.definitions.filter(isUseCaseDefinitionNode);
+
+    for (const useCase of useCases) {
+      this.visit(useCase);
+    }
+  }
+
+  visitProfileHeaderNode(_node: ProfileHeaderNode): void {
+    throw new Error('not implemented');
+  }
+
+  visitUseCaseDefinitionNode(node: UseCaseDefinitionNode): void {
+    this.currentUseCase = node.useCaseName;
+
+    if (node.examples) {
+      for (const example of node.examples) {
+        this.visit(example.value);
+      }
+    }
+
+    this.currentUseCase = undefined;
+  }
+
+  visitUseCaseSlotDefinition(
+    node: UseCaseSlotDefinitionNode<ProfileASTNode>
+  ): void {
+    this.visit(node.value);
+  }
+
+  visitUseCaseExampleNode(node: UseCaseExampleNode): void {
+    if (node.input) {
+      this.currentStructure = this.profileOutput?.usecases.find(
+        usecase => usecase.useCaseName === this.currentUseCase
+      )?.input;
+      this.slotType = ValidationIssueSlot.INPUT;
+
+      this.visit(node.input);
+
+      this.slotType = undefined;
+      this.currentStructure = undefined;
+    }
+
+    if (node.result) {
+      this.currentStructure = this.profileOutput?.usecases.find(
+        usecase => usecase.useCaseName === this.currentUseCase
+      )?.result;
+      this.slotType = ValidationIssueSlot.RESULT;
+
+      this.visit(node.result);
+
+      this.slotType = undefined;
+      this.currentStructure = undefined;
+    }
+
+    if (node.asyncResult) {
+      this.currentStructure = this.profileOutput?.usecases.find(
+        usecase => usecase.useCaseName === this.currentUseCase
+      )?.async;
+      this.slotType = ValidationIssueSlot.ASYNCRESULT;
+
+      this.visit(node.asyncResult);
+
+      this.slotType = undefined;
+      this.currentStructure = undefined;
+    }
+
+    if (node.error) {
+      this.currentStructure = this.profileOutput?.usecases.find(
+        usecase => usecase.useCaseName === this.currentUseCase
+      )?.error;
+      this.slotType = ValidationIssueSlot.ERROR;
+
+      this.visit(node.error);
+
+      this.slotType = undefined;
+      this.currentStructure = undefined;
+    }
+  }
+
+  visitComlinkPrimitiveLiteralNode(node: ComlinkPrimitiveLiteralNode): boolean {
+    if (!this.slotType) {
+      throw new Error('No slot type defined');
+    }
+
+    if (!this.currentStructure) {
+      this.warnings.push({
+        kind: 'useCaseSlotNotFound',
+        context: {
+          path: this.getPath(node),
+          slot: this.slotType,
+          actual: node,
+        },
```

Comment:
It can happen when profile doesn't define input/result/error, but it's used in example. Even though `this.currentStructure` is assigned [here](https://github.com/superfaceai/parser/blob/51b5c289b9e95471e2aff8706636847e5a9eed65/src/interpreter/example-validator.ts#L158-L160), it can be `undefined`.

Reply:
Right, I didn't realize that. Ok then.